neurovault/apps/statmaps/models.py

```diff
@@ -347,3 +351,33 @@ class Atlas(Image):
 
     class Meta:
         verbose_name_plural = "Atlases"
+
+
+class Similarity(models.Model):
```

Comment:
We can query just the value without a problem (
https://docs.djangoproject.com/en/1.7/ref/models/querysets/#django.db.models.query.QuerySet.values),
but we will not be able to avoid a JOIN operation and we will be storying
potentially millions of IRI strings. The latter can be solved by storying
similarity metric and transformation types is separate tables (that would
however add extra two JOINs when querying).

On Mon, Feb 2, 2015 at 10:08 AM, Tal Yarkoni <notifications@github.com>
wrote:

> In neurovault/apps/statmaps/models.py
> <https://github.com/NeuroVault/NeuroVault/pull/111#discussion_r23944043>:
>
> > @@ -347,3 +351,33 @@ class Atlas(Image):
> >
> >      class Meta:
> >          verbose_name_plural = "Atlases"
> > +
> > +
> > +class Similarity(models.Model):
>
> Not sure how Django handles associations, but it seems reasonable to keep
> this around as long as it's lazy loaded. If it's only used sporadically to
> look up the details of the similarity metric (e.g., once per results page).
> But it definitely shouldn't be eager-loaded every time a Comparison record
> is retrieved.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NeuroVault/NeuroVault/pull/111/files#r23944043>.
>